### PR TITLE
Update

### PR DESCRIPTION
**Pull request by bbhtt**
Updates the SDK to 41, perl to 5.34.1 and Hexchat to 2.16.1

### CODE REVIEW

**Comment by bbhtt** [comment]
@TingPing If you are around please stop the above builds, looks like they are stuck on missing dependencies and I don't have access. 

Also any idea why [91195](https://flathub.org/builds/#/builders/32/builds/91195) is failing with `Package 'gtk+-2.0', required by 'virtual:world', not found` ? I added gtk2 from the submodule. Also `-Dwith-gtk=false` and `-Dwith-dbus=false` is set, not sure why they are being required.

File: io.github.Hexchat.Plugin.Perl.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "branch": "20.08",
```
Comment (TingPing) [inline io.github.Hexchat.Plugin.Perl.json]: branch should be updated to `21.08`.

The main `io.github.Hexchat` package needs to have this set as the branch name for the extension.

Comment (TingPing) [reply io.github.Hexchat.Plugin.Perl.json]: https://github.com/flathub/io.github.Hexchat/commit/736e3ce05f4d61046b2541b8e4c46fba25961180

**Comment by TingPing** [comment]
You should not be building GTK or dbus-glib at all. HexChat should build without the gtk frontend and if it doesn't its a bug in its build system.

Note that I don't think the Perl plugin has changed at all in many releases so the old one is probably fine.

File: io.github.Hexchat.Plugin.Perl.json (hunk 1 above)
Comment (bbhtt) [reply io.github.Hexchat.Plugin.Perl.json]: Thanks.

**Comment by bbhtt** [comment]
> You should not be building GTK or dbus-glib at all. HexChat should build without the gtk frontend and if it doesn't its a bug in its build system.

It doesn't, 91192 was failing due to missing dbus-binding-tool:

```
Configuring org.hexchat.service.service using configuration
Program dbus-binding-tool found: NO
../src/common/dbus/meson.build:38:0: ERROR: Program 'dbus-binding-tool' not found
A full log can be found at /run/build/hexchat/_flatpak_build/meson-logs/meson-log.txt
FB: host_command_exited_cb 3856692 256
]2;flatpak-builder: Cleanup hexchatError: module hexchat: Child process exited with code 1
```

so I added it from the shared module but then  91194 fails with missing gtk+-2.0:

```
Found CMake: /usr/bin/cmake (3.22.3)
Run-time dependency gtk+-2.0 found: NO (tried pkgconfig and cmake)
../src/fe-gtk/meson.build:31:0: ERROR: Dependency "gtk+-2.0" not found, tried pkgconfig and cmake
A full log can be found at /run/build/hexchat/_flatpak_build/meson-logs/meson-log.txt
FB: host_command_exited_cb 521293 256
]2;flatpak-builder: Cleanup hexchatError: module hexchat: Child process exited with code 1
```
Edit. Looks like the flags were renamed here https://github.com/hexchat/hexchat/commit/cbb0927a7a9113d3b6b772e7b0566752dd54e6dd

**Comment by bbhtt** [comment]
[91352](https://flathub.org/builds/#/builders/32/builds/91352) fails with `install: cannot stat 'data/misc/io.github.Hexchat.Plugin.Perl.metainfo.xml': No such file or directory`, but I added this buildflag https://github.com/hexchat/hexchat/blob/master/meson_options.txt#L33. It fails similarly with `-Dwith-perl=/bin/perl` Any idea?

**Comment by bbhtt** [comment]
Ok I think I found the issue, I forgot to change it here as well https://github.com/flathub/io.github.Hexchat.Plugin.Perl/blob/master/io.github.Hexchat.Plugin.Perl.json#L80, tried building with that change locally and it built fine. This PR got messed up a bit I will continue in https://github.com/flathub/io.github.Hexchat.Plugin.Perl/pull/6

**Pull request closed without merging**